### ISSUE TITLE
quest can't complete

### ISSUE THREAD

**Issue by SimonChenyongxing** [TrinityCore/TrinityCore#26798]
### Description

quest 'a Delicate touch' can't be complete.

### Expected behaviour

Garm attackers slain always be 0.

### Steps to reproduce the problem

accept the quest

### Branch

3.3.5

### TC rev. hash/commit

 a47926ceab7091dd0c06b6bb16a0d1f64bd30a63

### Operating system

Windows 7 x64

### Custom changes

no

**Comment by dr-j** [comment]
Also change on update to IC as otherwise these landmines always explode 1 second after been placed when item clearly states these should last upto 5 minutes and should not detonate by themselves
~~~sql
UPDATE `smart_scripts` SET `event_type`='0', `event_flags`='1', `event_param1`='1000', `event_param2`='1000', `event_param3`='0', `event_param4`='0', `comment`='Improved Land Mine - IC  - Cast \'Detonation\'' WHERE  `entryorguid`=29475 AND `source_type`=0 AND `id`=1 AND `link`=2;
UPDATE `smart_scripts` SET `comment`='Improved Land Mine - IC  - Despawn In 500 ms' WHERE  `entryorguid`=29475 AND `source_type`=0 AND `id`=2 AND `link`=0;

**Comment by SimonChenyongxing** [comment]
Is your sql in the latest TDB?   Did  I miss some sql file?
Or you will update into TDB later?

**Comment by dr-j** [comment]
No since only noticed that was using nearest player to give kill credit when this issue was raised yesterday

**Comment by dr-j** [comment]
What’s in tdb id the commit I linked

**Comment by Aokromes** [comment]
i have seen this quest completed 4d ago.

**Comment by Faq** [comment]
His rev is 28d old, so might be that reason?

**Comment by Aokromes** [comment]
> His rev is 28d old, so might be that reason?

no, i have seen it completed with 99% success across years.

**Comment by dr-j** [comment]
Its simple the landmines need to deliver killing blow to npcs if you just slay them you wont get credit, also worth pointing out that unless you pre-damage mobs you will have to use 2 landmines to kill them as explosion from single mine leaves them with 10-20% hp and in order to get credit killing blow must be caused by mine.

**Comment by SimonChenyongxing** [comment]
It is resolved. thanks

**Comment by offl** [comment]
It does work but in a weird way. It looks like mobs dies on their own under specific conditions, can't explain that. Anyways, I rescripted it but for some reason player receives 2 credits for one mob instead of 1. Turned out it happens even with old script and even if you remove everything related to quest credit. Looks like something redirects kill credit in core or it's handled somewhere else. I already hate everything related to that quest so don't wanna dig further. Gonna open another issue after push
